### ISSUE TITLE
control_path and control_path_dir settings uncorrectly documented

### ISSUE THREAD

**Issue by zeitounator** [ansible/ansible#75309]
### Summary

Reading the latest doc I came accros the following setting:
https://docs.ansible.com/ansible/latest/reference_appendices/config.html#persistent-control-path-dir

This states that in the `ansible.cfg` file, the `control_path_dir` setting should appear under the `[persistent_connection]` section.

Now looking at the example file from source, [this setting](https://github.com/ansible/ansible/blob/4d78b58540dafd818a5e75ec390f9f03f5367ed9/examples/ansible.cfg#L387) appears in the `[ssh_connection]` section which seems to always have been the case (checked up to version 1.5...). Trying locally confirms this is the correct place.

Moreover, there is an other config parameter in that same section called `control_path` which does not appear at all on the documentation page.

Expected: both settings are documented and aligned in the correct configuration section.

### Issue Type

Documentation Report

### Component Name

ansible.cfg

### Ansible Version

```console
$ ansible --version
ansible [core 2.11.2] 
  config file = None
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/user/lib/python3.8/dist-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Jun  2 2021, 10:49:15) [GCC 9.4.0]
  jinja version = 2.11.3
  libyaml = True
```

### Configuration

```console
$ ansible-config dump --only-changed
DEFAULT_VAULT_IDENTITY_LIST(env: ANSIBLE_VAULT_IDENTITY_LIST) = [..... REDACTED ......]
LOCALHOST_WARNING(env: ANSIBLE_LOCALHOST_WARNING) = False
```

### OS / Environment

Ubuntu 20.04

### Additional Information

N/A

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* [`examples/ansible.cfg`](https://github.com/ansible/ansible/blob/devel/examples/ansible.cfg)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by toydarian** [comment]
Ok, I think, I figured it out.  
The documentation is actually generated from `lib/ansible/config/base.py`, so it should be inherently correct.

What happened, is that all `ANSIBLE_SSH*` options were moved to the `ssh` plugin:
https://github.com/ansible/ansible/compare/v2.10.0...v2.11.0

And that is why those options don't show up in the documentation.

Apart from that, the `PERSISTENT_CONTROL_PATH_DIR` option still exists. I'm pretty sure that is redundant and should be removed.
